src/lib.rs

```diff
@@ -411,12 +417,14 @@ impl<Q: ?Sized, K: Borrow<Q>> Borrow<KeyRef<Q>> for Key<K> {
 struct CLruNode<K, V> {
     key: Key<K>,
     value: V,
+    weight: usize,
 }
 
 /// An LRU cache with constant time operations.
 pub struct CLruCache<K, V, S = RandomState> {
     lookup: HashMap<Key<K>, usize, S>,
     storage: FixedSizeList<CLruNode<K, V>>,
+    weight: usize,
```

Comment:
👍 

Curious to see how API evolves when `max_weight` comes in

Reply:
So I am not sure I want to add another limit. The way I think it should work is that the `capacity` *is* the `max_weight` you are talking about.

We might want to differentiate the concept of pre-allocation and actual maximum capacity of the cache though.
It would require a new constructor method like `with_capacity_and_pre_allocated_nodes()` but with a better name obviously. That change, if possible, should probably be added first in a separate PR. I will think about it.

Reply:
I'm happy if only a single weight limit is needed.

> We might want to differentiate the concept of pre-allocation and actual maximum capacity of the cache though.

If that works, great. My impression was that the codebase heavily depends on the knowledge of element counts in multiple places.

Reply:
I have successfully split the notion of capacity and preallocated memory in https://github.com/marmeladema/clru-rs/pull/31